### PR TITLE
network: add mechanism to configure default UseDomains= setting

### PR DESCRIPTION
**Pull request by henryli001**
This PR adds a new parameter to /etc/systemd/networkd.conf to allow controlling the default value of the UseDomain= parameter (for both dhcp v4 and v6).

### CODE REVIEW

**Comment by yuwata** [comment]
Conceptually good, but the CIs are red:
```
======================================================================
FAIL: test_dhcp_client_default_use_domains (__main__.NetworkdDHCPClientTests.test_dhcp_client_default_use_domains)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/./test/test-network/systemd-networkd-tests.py", line 6876, in test_dhcp_client_default_use_domains
    check(self, True, True)
  File "/build/./test/test-network/systemd-networkd-tests.py", line 6858, in check
    self.assertIn('example.com', output)
AssertionError: 'example.com' not found in 'Link 69 (veth99):'
```

File: src/network/networkd-gperf.gperf [hunk 1]
```diff
@@ -30,8 +30,10 @@ Network.RouteTable,                      config_parse_route_table_names,
 Network.IPv4Forwarding,                  config_parse_tristate,                  0,          offsetof(Manager, ip_forwarding[0])
 Network.IPv6Forwarding,                  config_parse_tristate,                  0,          offsetof(Manager, ip_forwarding[1])
 Network.IPv6PrivacyExtensions,           config_parse_ipv6_privacy_extensions,   0,          offsetof(Manager, ipv6_privacy_extensions)
+DHCPv4.DefaultUseDomains,                config_parse_default_dhcp_use_domains,  0,          offsetof(Manager, dhcp_use_domains)
```
Comment (yuwata) [inline src/network/networkd-gperf.gperf]: Please use `UseDomains`, that is, please drop `Default`.

File: src/network/networkd-gperf.gperf [hunk 2]
```diff
@@ -30,8 +30,10 @@ Network.RouteTable,                      config_parse_route_table_names,
...
 DHCPv4.DUIDType,                         config_parse_duid_type,                 0,          offsetof(Manager, dhcp_duid)
 DHCPv4.DUIDRawData,                      config_parse_duid_rawdata,              0,          offsetof(Manager, dhcp_duid)
+DHCPv6.DefaultUseDomains,                config_parse_default_dhcp_use_domains,  0,          offsetof(Manager, dhcp6_use_domains)
```
Comment (yuwata) [inline src/network/networkd-gperf.gperf]: Also here.

File: src/network/networkd-manager.c [hunk 3]
```diff
@@ -599,7 +599,9 @@ int manager_new(Manager **ret, bool test_mode) {
                 .manage_foreign_nexthops = true,
                 .ethtool_fd = -EBADF,
                 .persistent_storage_fd = persistent_storage_open(),
+                .dhcp_use_domains = DHCP_USE_DOMAINS_NO,
                 .dhcp_duid.type = DUID_TYPE_EN,
+                .dhcp6_use_domains = DHCP_USE_DOMAINS_NO,
```
Comment (yuwata) [inline src/network/networkd-manager.c]: `DHCP_USE_DOMAINS_NO` is zero, thus, these lines are not necessary. Please drop them.

**Review by yuwata — CHANGES_REQUESTED** [review]
Please update the man pages.

File: man/networkd.conf.xml [hunk 4]
```diff
@@ -276,9 +287,10 @@ DUIDRawData=00:00:ab:11:f9:2a:c2:77:29:f9:5c:00</programlisting>
       <varlistentry>
         <term><varname>DUIDType=</varname></term>
         <term><varname>DUIDRawData=</varname></term>
+        <term><varname>DefaultUseDomains=</varname></term>
         <listitem><para>As in the [DHCPv4] section.</para>
 
-        <xi:include href="version-info.xml" xpointer="v249"/></listitem>
+        <xi:include href="version-info.xml" xpointer="v256"/></listitem>
```
Comment (yuwata) [inline man/networkd.conf.xml:299]: Please do not merge the entry with `DUIDType=` and `DUIDRawData=`, to keep the version info for them.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/network/networkd-gperf.gperf (hunk 1 above)
Comment (henryli001) [reply src/network/networkd-gperf.gperf]: Is there a reason why having "Default" in the naming is not suggested? Using a different name can better separate this parameter from the existing "UseDomains" field.

Comment (yuwata) [reply src/network/networkd-gperf.gperf]: Why the new setting needs to be separated from `UseDomains` in .network file?
Yes, for PID1, we have bunch of DefaultXYZ setting. But that's quite redundant I think, and other existing settings, e.g. `IPv4Forwarding=` or so does not have `Default` prefix.

File: man/networkd.conf.xml [hunk 5]
```diff
@@ -256,6 +256,17 @@ DUIDRawData=00:00:ab:11:f9:2a:c2:77:29:f9:5c:00</programlisting>
           <xi:include href="version-info.xml" xpointer="v230"/>
         </listitem>
       </varlistentry>
+
+      <varlistentry>
+        <term><varname>UseDomains=</varname></term>
+          <listitem><para>Specifies the default value for <varname>UseDomains=</varname>.
```
Comment (yuwata) [inline man/networkd.conf.xml]: `default value for <varname>UseDomains=</varname>` -> `default value for per-network <varname>UseDomains=</varname>` is better?

File: src/network/networkd-gperf.gperf [hunk 6]
```diff
@@ -30,8 +30,10 @@ Network.RouteTable,                      config_parse_route_table_names,
 Network.IPv4Forwarding,                  config_parse_tristate,                  0,          offsetof(Manager, ip_forwarding[0])
 Network.IPv6Forwarding,                  config_parse_tristate,                  0,          offsetof(Manager, ip_forwarding[1])
 Network.IPv6PrivacyExtensions,           config_parse_ipv6_privacy_extensions,   0,          offsetof(Manager, ipv6_privacy_extensions)
+DHCPv4.UseDomains,                config_parse_default_dhcp_use_domains,  0,          offsetof(Manager, dhcp_use_domains)
 DHCPv4.DUIDType,                         config_parse_duid_type,                 0,          offsetof(Manager, dhcp_duid)
 DHCPv4.DUIDRawData,                      config_parse_duid_rawdata,              0,          offsetof(Manager, dhcp_duid)
+DHCPv6.UseDomains,                config_parse_default_dhcp_use_domains,  0,          offsetof(Manager, dhcp6_use_domains)
```
Comment (yuwata) [inline src/network/networkd-gperf.gperf]: Indentation.

**Review by yuwata — CHANGES_REQUESTED** [review]
Please also update systemd.network.xml to mention that the default value of `UseDomains=` is given by networkd.conf.

File: man/systemd.network.xml [hunk 7]
```diff
@@ -2570,7 +2570,8 @@ NFTSet=prefix:netdev:filter:eth_ipv4_prefix</programlisting>
           effect of the <option>Domains=</option> setting. If set to <option>route</option>, the domain name
           received from the DHCP server will be used for routing DNS queries only, but not for searching,
           similarly to the effect of the <option>Domains=</option> setting when the argument is prefixed with
-          <literal>~</literal>. Defaults to false.</para>
+          <literal>~</literal>. The default value is provided by the setting of <varname>UseDomains=</varname> in 
+          networkd.conf. Defaults to false.</para>
```
Comment (yuwata) [inline man/systemd.network.xml]: How about the following, (copied from `IPv6PrivacyExtensions=`):
```
When unspecified, the value specified in the same setting in
<citerefentry><refentrytitle>networkd.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry>,
which defaults to <literal>no</literal>, will be used.
```

**Comment by yuwata** [comment]
@henryli001 As you can see https://github.com/systemd/systemd/blob/9944629eeeeadd02b7e79592d213707696ea9347/test/test-network/systemd-networkd-tests.py#L5-L16, you can run the test locally with something like the following:
```
sudo ./systemd-networkd-tests.py --build-dir=/path/to/build NetworkdDHCPClientTests.test_dhcp_client_default_use_domains
```

File: src/network/networkd-dhcp-common.c [hunk 8]
```diff
@@ -632,6 +632,36 @@ int config_parse_dhcp_use_domains(
         return 0;
 }
 
+int config_parse_default_dhcp_use_domains(
```
Comment (yuwata) [inline src/network/networkd-dhcp-common.c]: You can use `DEFINE_CONFIG_PARSE_ENUM_FULL()` macro to define such a simple conf parser. See shared/conf-parser.h

File: test/test-network/systemd-networkd-tests.py [hunk 9]
```diff
@@ -6839,6 +6839,48 @@ def check(self, ipv4, ipv6):
...
         check(self, False, False)
+    
+    def test_dhcp_client_default_use_domains(self):        
+        def check(self, ipv4, ipv6):
+            os.makedirs(networkd_conf_dropin_dir, exist_ok=True)
+            with open(os.path.join(networkd_conf_dropin_dir, 'default_use_domains.conf'), mode='w', encoding='utf-8') as f:
+                f.write('[DHCPv4]\nUseDomains=')
+                f.write('yes' if ipv4 else 'no')
+                f.write('\n[DHCPv6]\nUseDomains=')
+                f.write('yes' if ipv6 else 'no')
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: missing `\n` at the end?

File: test/test-network/systemd-networkd-tests.py [hunk 10]
```diff
@@ -6839,6 +6839,48 @@ def check(self, ipv4, ipv6):
...
+            
+            check_output('systemctl daemon-reload')
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: This is not necessary, as the config is for networkd, rather than PID1.

File: test/test-network/systemd-networkd-tests.py [hunk 11]
```diff
@@ -6839,6 +6839,48 @@ def check(self, ipv4, ipv6):
...
+            restart_networkd()
+            check_output('systemctl restart systemd-resolved.service')
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: This is not necessary. resolved automatically update DNS related settings when networkd acquires a DHCP lease.

File: test/test-network/systemd-networkd-tests.py [hunk 12]
```diff
@@ -6839,6 +6839,48 @@ def check(self, ipv4, ipv6):
...
+            if ipv4 or ipv6:
+                self.assertIn('example.com', output)
+            else:
+                self.assertNotIn('example.com', output)
+            
+            remove_networkd_conf_dropin('default_use_domains.conf')
+
+        copy_network_unit('25-veth.netdev', '25-dhcp-server-veth-peer.network', '25-dhcp-client.network', copy_dropins=False)
+
+        start_networkd()
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: It is not necessary to start networkd here, as it will be anyway restarted.

File: test/test-network/systemd-networkd-tests.py [hunk 13]
```diff
@@ -6839,6 +6839,48 @@ def check(self, ipv4, ipv6):
...
+        self.wait_online('veth-peer:carrier')
+        start_dnsmasq('--dhcp-option=option:dns-server,192.168.5.1',
+                      '--dhcp-option=option6:dns-server,[2600::1]',
+                      '--dhcp-option=option:domain-search,example.com',
+                      '--dhcp-option=option6:domain-search,example.com')
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: Move these lines after `restart_networkd()` in `check()`.

File: test/test-network/systemd-networkd-tests.py [hunk 14]
```diff
@@ -6839,6 +6839,48 @@ def check(self, ipv4, ipv6):
...
+            self.wait_address('veth99', r'inet6 2600::[0-9a-f]*/128 scope global (dynamic noprefixroute|noprefixroute dynamic)', ipv='-6')
+
+            output = resolvectl('domain', 'veth99')
+            print(output)
+            if ipv4 or ipv6:
+                self.assertIn('example.com', output)
+            else:
+                self.assertNotIn('example.com', output)
+            
+            remove_networkd_conf_dropin('default_use_domains.conf')
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py:6879]: Then, kill dnsmasq before calling next `check()`.

File: test/test-network/systemd-networkd-tests.py [hunk 15]
```diff
@@ -6839,6 +6839,48 @@ def check(self, ipv4, ipv6):
...
+        self.wait_online('veth-peer:carrier')
+        start_dnsmasq('--dhcp-option=option:dns-server,192.168.5.1',
+                      '--dhcp-option=option6:dns-server,[2600::1]',
+                      '--dhcp-option=option:domain-search,example.com',
+                      '--dhcp-option=option6:domain-search,example.com')
+
+        check(self, True, True)
+        check(self, True, False)
+        check(self, False, True)
+        check(self, False, False)
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py:6885]: Not a matter, but I prefer `self.check()` rather than `check(self)`.

File: test/test-network/systemd-networkd-tests.py [hunk 16]
```diff
@@ -6839,6 +6839,48 @@ def check(self, ipv4, ipv6):
...
+            restart_networkd()
+            check_output('systemctl restart systemd-resolved.service')
+
+            self.wait_online('veth99:routable')
+
+            # link becomes 'routable' when at least one protocol provide an valid address. Hence, we need to explicitly wait for both addresses.
+            self.wait_address('veth99', r'inet 192.168.5.[0-9]*/24 metric 1024 brd 192.168.5.255 scope global dynamic', ipv='-4')
+            self.wait_address('veth99', r'inet6 2600::[0-9a-f]*/128 scope global (dynamic noprefixroute|noprefixroute dynamic)', ipv='-6')
+
+            output = resolvectl('domain', 'veth99')
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: This may be racy, as even if networkd acquires DHCPv{4,6} leases, resolved may not be read DNS related settings yet. Hence, below check needs to be run several times something like the following:
```python
for _ in range(20):
        output = resolvectl('domain', 'veth99')
        if ipv4 or ipv6:
                if 'example.com' in output:
                        break
        else:
                if 'example.com' not in output:
                        break
        time.sleep(0.5)
else:
        print(output)
        self.fail('unexpected domain setting in resolved...')
```

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by yuwata** [comment]
By the way, please prefix the commit title with `network:`

**Comment by henryli001** [comment]
> @henryli001 As you can see
> 
> https://github.com/systemd/systemd/blob/9944629eeeeadd02b7e79592d213707696ea9347/test/test-network/systemd-networkd-tests.py#L5-L16
> 
> , you can run the test locally with something like the following:
> ```
> sudo ./systemd-networkd-tests.py --build-dir=/path/to/build NetworkdDHCPClientTests.test_dhcp_client_default_use_domains
> ```

I followed the instructions to run the test locally within a QEMU image, but ended up running into an error saying that the directory is not a systemd source tree. I think is very likely an issue with my local setup even though I double checked all the commands bring run against the instruction. Need to re-investigate on this front.

**Comment by yuwata** [comment]
> > @henryli001 As you can see
> > https://github.com/systemd/systemd/blob/9944629eeeeadd02b7e79592d213707696ea9347/test/test-network/systemd-networkd-tests.py#L5-L16
> > 
> > , you can run the test locally with something like the following:
> > ```
> > sudo ./systemd-networkd-tests.py --build-dir=/path/to/build NetworkdDHCPClientTests.test_dhcp_client_default_use_domains
> > ```
> 
> I followed the instructions to run the test locally within a QEMU image, but ended up running into an error saying that the directory is not a systemd source tree. I think is very likely an issue with my local setup even though I double checked all the commands bring run against the instruction. Need to re-investigate on this front.

I do not think it is necessary to run the test on QEMU. You can safely run the test on your local machine.

> an error saying that the directory is not a systemd source tree.

If you'd like to run the test on QEMU, then the easiest way is building systemd in the QEMU image, and run the test.

File: man/systemd.network.xml [hunk 17]
```diff
@@ -2570,7 +2570,9 @@ NFTSet=prefix:netdev:filter:eth_ipv4_prefix</programlisting>
           effect of the <option>Domains=</option> setting. If set to <option>route</option>, the domain name
           received from the DHCP server will be used for routing DNS queries only, but not for searching,
           similarly to the effect of the <option>Domains=</option> setting when the argument is prefixed with
-          <literal>~</literal>. Defaults to false.</para>
+          <literal>~</literal>. When unspecified, the value specified in the same setting in
+          <citerefentry><refentrytitle>networkd.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry>,
+          which defaults to <literal>no</literal>, will be used.</para>
```
Comment (ddstreetmicrosoft) [inline man/systemd.network.xml:2575]: Don't include what the default value of the manager's parameter is; that is redundant and also could easily be missed by downstream distros who change the default value of the manager's parameter (and only change the doc for `networkd.conf`). Here, just state that the default value (of the network's parameter) is the value of the manager's parameter (i.e. leave out `which defaults to no`)

File: src/network/networkd-gperf.gperf [hunk 18]
```diff
@@ -30,8 +30,10 @@ Network.RouteTable,                      config_parse_route_table_names,
 Network.IPv4Forwarding,                  config_parse_tristate,                  0,          offsetof(Manager, ip_forwarding[0])
 Network.IPv6Forwarding,                  config_parse_tristate,                  0,          offsetof(Manager, ip_forwarding[1])
 Network.IPv6PrivacyExtensions,           config_parse_ipv6_privacy_extensions,   0,          offsetof(Manager, ipv6_privacy_extensions)
+DHCPv4.UseDomains,                       config_parse_default_dhcp_use_domains,  0,          offsetof(Manager, dhcp_use_domains)
```
Comment (ddstreetmicrosoft) [inline src/network/networkd-gperf.gperf:33]: Personally I would use `DefaultUseDomains` here, as `system.conf` does for most of its default parameters.

Comment (henryli001) [reply src/network/networkd-gperf.gperf:33]: I started with `DefaultUseDomains` but @yuwata raised a point where having a separate naming might be redundant (including the existing configs). Perhaps something we can further discuss here to determine what the go-to pattern should look like?

Comment (YHNdnzj) [reply src/network/networkd-gperf.gperf:33]: I agree with @yuwata that `UseDomains=` should be the way to go. Not only because `Default` is redundant but also that `networkd.conf` is already using such naming scheme.

File: man/systemd.network.xml [hunk 19]
```diff
@@ -2570,7 +2570,8 @@ NFTSet=prefix:netdev:filter:eth_ipv4_prefix</programlisting>
           effect of the <option>Domains=</option> setting. If set to <option>route</option>, the domain name
           received from the DHCP server will be used for routing DNS queries only, but not for searching,
           similarly to the effect of the <option>Domains=</option> setting when the argument is prefixed with
-          <literal>~</literal>. Defaults to false.</para>
+          <literal>~</literal>. When unspecified, the value specified in the same setting in
+          <citerefentry><refentrytitle>networkd.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry>, will be used.</para>
```
Comment (yuwata) [inline man/systemd.network.xml]: I'd like to keep `which defaults to <literal>no</literal>`. Otherwise, users need to jump to networkd.conf(5) and check the default explained there.

Of course, as commented, distribution may provide netowkrd.conf or its drop-in with non-default value, but that's not a problem here, as that's not the default value of networkd. The statement explains the default value, rather than the distro configuration, and does not contradict with the value explained in networkd.conf(5).

In short, please re-add the statement about the default value.

File: src/network/networkd-dhcp-common.c [hunk 20]
```diff
@@ -632,6 +632,8 @@ int config_parse_dhcp_use_domains(
         return 0;
 }
 
+DEFINE_CONFIG_PARSE_ENUM(config_parse_default_dhcp_use_domains, dhcp_use_domains, DHCPUseDomains, "Failed to parse default UseDomains=")
```
Comment (yuwata) [inline src/network/networkd-dhcp-common.c]: `default` in the error message is redundant. Please drop.

File: test/test-network/systemd-networkd-tests.py [hunk 21]
```diff
@@ -6839,6 +6839,51 @@ def check(self, ipv4, ipv6):
         check(self, True, False)
         check(self, False, True)
         check(self, False, False)
+    
+    def test_dhcp_client_default_use_domains(self):        
+        def check(self, ipv4, ipv6):
+            os.makedirs(networkd_conf_dropin_dir, exist_ok=True)
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: Please use `mkdir_p(networkd_conf_dropin_dir)`.

File: test/test-network/systemd-networkd-tests.py [hunk 22]
```diff
@@ -6839,6 +6839,51 @@ def check(self, ipv4, ipv6):
...
+
+            self.wait_online('veth99:routable')
+
+            # link becomes 'routable' when at least one protocol provide an valid address. Hence, we need to explicitly wait for both addresses.
+            self.wait_address('veth99', r'inet 192.168.5.[0-9]*/24 metric 1024 brd 192.168.5.255 scope global dynamic', ipv='-4')
+            self.wait_address('veth99', r'inet6 2600::[0-9a-f]*/128 scope global (dynamic noprefixroute|noprefixroute dynamic)', ipv='-6')
+
+            for _ in range(20):
+                output = resolvectl('domain', 'veth99')
+                print(output)
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: Not a matter, but print() here is noisy. And `print()` in the `else` branch duplicates the output.
So, please drop this, and keep `print()` in `else`.

File: test/test-network/systemd-networkd-tests.py [hunk 23]
```diff
@@ -6839,6 +6839,51 @@ def check(self, ipv4, ipv6):
...
         check(self, False, False)
+    
+    def test_dhcp_client_default_use_domains(self):        
+        def check(self, ipv4, ipv6):
+            os.makedirs(networkd_conf_dropin_dir, exist_ok=True)
+            with open(os.path.join(networkd_conf_dropin_dir, 'default_use_domains.conf'), mode='w', encoding='utf-8') as f:
+                f.write('[DHCPv4]\nUseDomains=')
+                f.write('yes' if ipv4 else 'no')
+                f.write('\n[DHCPv6]\nUseDomains=')
+                f.write('yes\n' if ipv6 else 'no\n')
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: Not a big matter, but this is slightly asymmetric. Please move `\n` before `[DHCPv6]` to the first `yes` or `no`.
```python
                f.write('[DHCPv4]\nUseDomains=')
                f.write('yes\n' if ipv4 else 'no\n')
                f.write('[DHCPv6]\nUseDomains=')
                f.write('yes\n' if ipv6 else 'no\n')```

File: test/test-network/systemd-networkd-tests.py (hunk 15 above)
Comment (yuwata) [reply test/test-network/systemd-networkd-tests.py:6885]: Not addressed? Yes. In the above another test, we have `check(self)`, but I really prefer `self.check()` style. Or, it does not work? I am not a python guru, so I am not sure...

**Review by yuwata — CHANGES_REQUESTED** [review]
Looks mostly good. Several minor cosmetic issues there.

**Comment by henryli001** [comment]
GitHub somehow does not allow me to quote your comment on the check(self) versus self.check() comment.

For this case, self.check() won't work since the `check()` function is only defined within the `test_dhcp_client_default_use_domains()` and not within the `NetworkdDHCPClientTests` definition and thus if doing self.check(), an error will be raised about the `NetworkdDHCPClientTests` object has no `check` attribute.

**Comment by yuwata** [comment]
> GitHub somehow does not allow me to quote your comment on the check(self) versus self.check() comment.
> 
> For this case, self.check() won't work since the `check()` function is only defined within the `test_dhcp_client_default_use_domains()` and not within the `NetworkdDHCPClientTests` definition and thus if doing self.check(), an error will be raised about the `NetworkdDHCPClientTests` object has no `check` attribute.

OK. Thanks.

**Review by yuwata — APPROVED** [review]
LGTM. Thank you! Nice work!

**Comment by rpigott** [comment]
This affects DHCP v4 and v6 but not ipv6ra? Is that intentional?

**Comment by yuwata** [comment]
No, it is ok to add that for RA when someone request that.

**Comment by evverx** [comment]
>  it is ok to add that for RA when someone request that

The documentation should be adjusted then. Right now it says that
> Specifies the default value for per-network UseDomains=

and it's misleading.

**Comment by rpigott** [comment]
So, I think you have the same misunderstanding as me: there are two options, one for DHCPv4 and DHCPv6 each. In that case I think yeah it's fine that we don't have an ipv6ra one, if a little odd.

For whatever reason when rebasing I only had a conflict in one area though, so I incorrectly thought there was only one option that applied to both, before I looked at the patch.

**Comment by evverx** [comment]
> there are two options, one for DHCPv4 and DHCPv6 each. In that case I think yeah it's fine that we don't have an ipv6ra one, if a little odd

I'm not sure what the point of this setting is then. I'd need to take a closer look probably.

**Comment by rpigott** [comment]
I opened #32273 which is a different kind of generalization, which I think might be more useful (and what I thought this was at first).

**Comment by evverx** [comment]
Looking at https://github.com/microsoft/azurelinux/pull/8741 I'm guessing there is no way there to set things granularly, reload configs and reconfigure specific interfaces so there should probably be a global switch that can be set in the global config (and that's probably where this PR came from). The problem is that `networkd` has to be restarted for those settings to kick in and it makes it even harder to figure out whether things are set or not.
